### PR TITLE
Specify path to be dropped

### PR DESCRIPTION
**Pull request by RamyChaabane**

### CODE REVIEW

File: test/e2e/drop_routes_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	time.Sleep(1 * time.Second)
+
+	// testing access to /favicon.ico -- should not have access
+	req, err = http.NewRequest("GET", "http://"+name+".example.com/favicon.ico", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resp, err = client.Do(req)
+	assert.Check(t, err == nil)
```
Comment (thomas-tacquet) [inline test/e2e/drop_routes_test.go]: assert.NilError(t, err)

File: test/e2e/drop_routes_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	_, err = clients.NetworkingClient.Ingresses.Patch(ctx, ing.Name, types.MergePatchType, patch, metav1.PatchOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// wait for the new config to be applied
+	time.Sleep(1 * time.Second)
+
+	// testing access to /favicon.ico -- should not have access
+	req, err = http.NewRequest("GET", "http://"+name+".example.com/favicon.ico", nil)
```
Comment (thomas-tacquet) [inline test/e2e/drop_routes_test.go:99]: "GET" = `http.MethodGet`

File: test/e2e/drop_routes_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = clients.NetworkingClient.Ingresses.Patch(ctx, ing.Name, types.MergePatchType, patch, metav1.PatchOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// wait for the new config to be applied
+	time.Sleep(1 * time.Second)
```
Comment (thomas-tacquet) [inline test/e2e/drop_routes_test.go]: `1 *` not necessary

File: pkg/generator/ingress_translator.go [hunk 4]
```diff
@@ -63,6 +64,16 @@ type IngressTranslator struct {
...
 }
 
+type DropRouteConfig struct {
+	// Path that will be dropped
+	Path string `json:"path"`
+}
+
+type DropRoutes struct {
+	// Configurations of routes that will be dropped
+	Routes []*DropRouteConfig `json:"routes"`
```
Comment (thomas-tacquet) [inline pkg/generator/ingress_translator.go]: I can't find puroposes of why DropRouteConfig is a pointer, imo should be a value

File: README.md [hunk 5]
```diff
@@ -216,14 +216,23 @@ ingress namespace:
...
 A good use case for this configuration is `DomainMapping with Websocket`
 
 Note: This annotation is an experimental/alpha feature. There is a known issue such as [issues/821](https://github.com/knative-sandbox/issues/821) and we may change the annotation name in the future.
 
+2. You might want to drop a specific route, for example you want to prevent your browser from requesting `/favicon.ico` automatically.
+To do that, you will need to annotate your Knative Service as following:
+```
+kubectl annotate service.serving.knative.dev/<knative_service_name> kourier.knative.dev/drop-routes='{"routes":[{"path":"/favicon.ico"}]}'
+```
+Notice that the object `routes` in the annotation is a list of map, so it's possible to add multiple paths following the same structure
```
Comment (thomas-tacquet) [inline README.md]:
```suggestion
Notice that the object `routes` in the annotation is a map of objects, so it's possible to add multiple paths following the same structure
```

File: test/e2e/drop_routes_test.go (hunk 2 above)
Comment (RamyChaabane) [reply test/e2e/drop_routes_test.go:99]: `http.MethodGet` is better, but I would like to keep the same pattern used in the other tests

File: pkg/generator/ingress_translator.go (hunk 4 above)
Comment (RamyChaabane) [reply pkg/generator/ingress_translator.go]: it's faster to have array of pointer than having an array of sturct

**Pull request closed without merging**